test/specs/utils/deepMerge.spec.js

```diff
@@ -6,7 +6,7 @@ describe('utils::deepMerge', function () {
     var b = {foo: 123};
     var c = {bar: 456};
 
-    deepMerge(a, b, c);
+    deepMerge({}, a, b, c);
```

Comment:
Can we keep the old interface, without the first argument as the target?

Reply:
In the following code we can keep the old interface without having to target the first parameter,Do you have a better idea?
```javascript
function deepMerge() {
  var target = {};
  // eslint-disable-next-line one-var
  var src, clone, copyIsArray;
  var idx = 0;
  if (typeof arguments[0] === 'boolean') {
    target = arguments[1] || {};
    idx = 2;
  }
  function assignValue(val, key) {
    src = target[key];
    if (isPlainObject(val) || (copyIsArray = isArray(val))) {
      if (copyIsArray) {
        copyIsArray = false;
        clone = src && isArray(src) ? src : [];
      } else {
        clone = src && isPlainObject(src) ? src : {};
      }
      target[key] = deepMerge(true, clone, val);
    } else {
      target[key] = val;
    }
  }
  for (; idx < arguments.length; idx++) {
    forEach(arguments[idx], assignValue);
  }
  return target;
}
```

Reply:
No, if I got an ideal solution, I will open a new PR. :)

Did you try the linked 2 similar pull request to get inspirations?